lib/rules/no-yield-to-default.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+const Rule = require('./base');
+
+const ERROR_MESSAGE = 'You should not yield to default';
+
+const DISALLOWED_PATHS = ['yield', 'has-block', 'has-block-params', 'hasBlock', 'hasBlockParams'];
+
+module.exports = class NoYieldToDefault extends Rule {
+  visitor() {
+    return {
+      MustacheStatement(node) {
+        if (DISALLOWED_PATHS.includes(node.path.original)) {
+          let isDefaultHashPair = node.hash.pairs.some((p) => {
+            return p.key === 'to' && p.value.value === 'default';
```

Comment:
Checking `.value.value` will error if the hash pair value is not a static string.

Reply:
Which means we should look out for dynamic values for named blocks like `to` ?

Reply:
That is in fact an error, but I think for the purpose of this rule you can just ignore that cause (don't blow up) and let something else handle it. (IIRC, the glimmer compiler already has a syntax error for that.)